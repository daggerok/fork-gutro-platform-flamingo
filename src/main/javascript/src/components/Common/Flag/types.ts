export type FlagProps = {
  country: string;
  isSmall?: boolean;
}