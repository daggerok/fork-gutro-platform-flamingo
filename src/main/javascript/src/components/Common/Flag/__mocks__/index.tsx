const mockedComponent = "mock-Flag";
export default mockedComponent;