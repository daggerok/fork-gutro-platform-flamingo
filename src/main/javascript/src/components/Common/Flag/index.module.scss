.flagIcon {
  font-size: 1.5em;
  box-shadow: 1px 1px 1px #999 !important;
}

.smallIcon {
  font-size: 1em;
}