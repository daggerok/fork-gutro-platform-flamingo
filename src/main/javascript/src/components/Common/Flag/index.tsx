import React from 'react';
import classnames from 'classnames';
import 'flag-icon-css/css/flag-icon.css';
import { FlagProps } from './types';

import styles from './index.module.scss';

const Flag: React.FC<FlagProps> = ({ country, isSmall }: FlagProps) => {
  if (!country) return null;

  return (
    <span
      data-testid="flag-component"
      className={classnames(
        styles.flagIcon,
        `flag-icon flag-icon-${country.toLowerCase()}`,
        isSmall && styles.smallIcon
      )}
    />
  );
};

export default Flag;
