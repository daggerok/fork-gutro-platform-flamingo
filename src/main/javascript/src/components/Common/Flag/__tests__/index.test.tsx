import React from "react";
import { render, screen } from "@testing-library/react";

import Flag from "..";

let component: any = null;

describe("<Flag />", () => {
  it("by default", () => {
    const { asFragment } = render(<Flag country="mt" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("without countrry", () => {
    const { asFragment } = render(<Flag country={""} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("with small size", () => {
    const { asFragment } = render(<Flag country={"SE"} isSmall />);

    expect(asFragment()).toMatchSnapshot();
  });
});
