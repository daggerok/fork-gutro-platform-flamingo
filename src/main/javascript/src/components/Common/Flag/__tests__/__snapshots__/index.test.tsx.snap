// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flag /> by default 1`] = `
<DocumentFragment>
  <span
    class="flagIcon flag-icon flag-icon-mt"
    data-testid="flag-component"
  />
</DocumentFragment>
`;

exports[`<Flag /> with small size 1`] = `
<DocumentFragment>
  <span
    class="flagIcon flag-icon flag-icon-se smallIcon"
    data-testid="flag-component"
  />
</DocumentFragment>
`;

exports[`<Flag /> without countrry 1`] = `<DocumentFragment />`;
